### ISSUE TITLE
Ability to exclude/ignore certain warnings?

### ISSUE THREAD

**Issue by pdehaan** [mozilla/addons-linter#746]
Currently I'm using **addons-linter** to validate a few add-ons for Test Pilot experiments (and Test Pilot itself). I'm getting a few warnings, such as `LOW_LEVEL_MODULE` which says "Usage of low-level or non-SDK interface". It'd be nice if there was an option to ignore those types of warnings from the report since I know we're u sing low-level SDKs, and I don't need the warnings in my Travis-CI logs each time.

I tried searching the README, issues list, and code but didn't see anything obvious about ignoring certain warnings/notices/errors (but I'm notoriously bad at reading, so maybe I'm wrong).
Other projects I've used either have some .addonsrc config file or I can put a list of codes in the package.json under some key and the `addons-linter` CLI tool will check there.

Current output:

``` bash
$ ./node_modules/.bin/addons-linter testpilot/frontend/static/addon/addon.xpi -o text

Validation Summary:
errors          0
notices         0
warnings        4

WARNINGS:

Code               Message                Description                                File                   Line   Column
LOW_LEVEL_MODULE   Usage of low-level     Your add-on uses an interface which        index.js               14     22
                   or non-SDK interface   bypasses the high-level protections of
                                          the add-on SDK. This interface should be
                                          avoided, and its use may significantly
                                          complicate your review process
LOW_LEVEL_MODULE   Usage of low-level     Your add-on uses an interface which        lib/metrics.js         9      14
                   or non-SDK interface   bypasses the high-level protections of
                                          the add-on SDK. This interface should be
                                          avoided, and its use may significantly
                                          complicate your review process
LOW_LEVEL_MODULE   Usage of low-level     Your add-on uses an interface which        lib/metrics.js         14     16
                   or non-SDK interface   bypasses the high-level protections of
                                          the add-on SDK. This interface should be
                                          avoided, and its use may significantly
                                          complicate your review process
LOW_LEVEL_MODULE   Usage of low-level     Your add-on uses an interface which        lib/cookie-manager.…   7      18
                   or non-SDK interface   bypasses the high-level protections of     js
                                          the add-on SDK. This interface should be
                                          avoided, and its use may significantly
                                          complicate your review process


$ echo $?
0
```

**Comment by andymckay** [comment]
This is a dupe of #662

**Comment by tofumatt** [comment]
Well, not quite. This is asking about disabling/ignoring certain warnings generically; the other issue (#662) seems to be specific to that error and a false positive.

Though: we don't have official support for non-webextensions anymore, so really we should be ditching this error anyway 😄

**Comment by andymckay** [comment]
I think if #662 was fixed, @pdehaan might not be filing this one, but we'll let him make that decision :)

**Comment by tofumatt** [comment]
Heh, fair enough.

**Comment by pdehaan** [comment]
Yeah, ideally I'd like to just give **addons-linter** CLI an array of codes to ignore.
If I lint the activity-stream.xpi, I also get this warning, which I don't need to see every time I run the linter (since I know we're overriding things, because we're hardcore like that):

```bash
WARNINGS:

Code               Message                        Description                                                File                     Line   Column
...
TAB_URL_OVERRIDE   Extensions must not alter      Extensions must not alter user preferences such as the     lib/ActivityStreams.js   53     3
                   user preferences such as the   new tab URL without explicit user consent. Such changes
                   new tab URL without explicit   must also be reverted when the extension is disabled or
                   user consent.                  uninstalled.
```

**Comment by pdehaan** [comment]
Yeah, I don't think mine is a dupe since I'd like the ability to silence any arbitrary warning via a config file (which I think is a broader request than the linked #662).
🍻

**Comment by andymckay** [comment]
But you can't do that in a WebExtension either, it shouldn't even be making that check.

**Comment by pdehaan** [comment]
I think the only solution is to fix #662 and see if I go away. 😉

**Comment by gregglind** [comment]
+1 on this.

I like the log level stuff.  

It seems like a yaml based on eslintrc type stuff might make sense.

**Comment by gregglind** [comment]
for just detecting if there are any errors:

```bash
echo '{"errors":[1]}' | python -c "import fileinput; import json; import sys; sys.exit(int(bool(json.loads(''.join(fileinput.input('-')))['errors'])))"
echo '{"errors":[]}' | python -c "import fileinput; import json; import sys; sys.exit(int(bool(json.loads(''.join(fileinput.input('-')))['errors'])))"
```

**Comment by gregglind** [comment]
Peeps:  I have a working prototype for something useful on this:

https://gist.github.com/gregglind/61bf2c75bf57988d3e5ca8c26ba87c41

**Comment by EnTeQuAk** [comment]
general ideas to this: we do have a relatively good grouping of valid codes that can be raised. The problem here is more that the current constants do not have an explicit severity level (info, warning, error) defined for them. I wonder if they should. From what I see, messages are only raised with exactly the same severity all the time. I could be missing something though.

Ideas on that?

If we'd have something like

```javascript
export const MANIFEST_VERSION_INVALID = {
  code: 'MANIFEST_VERSION_INVALID',
  severity: WARNING,
  message: _('"manifest_version" in the manifest.json is not a valid value'),
  description: _('See https://mzl.la/20PenXl (MDN Docs) for more information.'),
  file: MANIFEST_JSON,
};
```
(note the `severity` here)

We can use all this information to add a easier system to exclude/ignore warnings. Does it make sense to introduce something like a `addons-linter.json` configuration file?
